run.sh

```diff
@@ -1,67 +1,121 @@
-#!/bin/bash -exv
-# This script simulates a ceremony where we generate the certificates from
-# https://community.letsencrypt.org/t/lets-encrypt-new-hierarchy-plans/125517
-export SOFTHSM2_CONF=$PWD/softhsm2.conf
-echo "directories.tokendir = $PWD/softhsm/" > $SOFTHSM2_CONF
+#!/bin/bash -e
+
+function usage() {
+    echo -e "USAGE:
+    This script simulates a ceremony where we generate new intermediate
+    certificates.
+
+    ./$(basename ${0}) [-h]
+        -h | Outputs this help text"
+}
+
+if [ "${1}" == "-h" ]; then
+    usage
+    # Be nice to those asking for help :)
+    exit 0
+fi
+
+if [ $# -ne 0 ]; then
+    usage
+    exit 1
+fi
 
 # see init-softhsm.sh for slot initialization
+export SOFTHSM2_CONF="${PWD}/softhsm2.conf"
+echo "directories.tokendir = ${PWD}/softhsm/" > $SOFTHSM2_CONF
 
-# Simulating intermediate HSM
-ceremony --config e1-key.yaml
-ceremony --config e2-key.yaml
-ceremony --config r3-key.yaml
-ceremony --config r4-key.yaml
-ceremony --config r3-cross-csr.yaml
-ceremony --config r4-cross-csr.yaml
+# Store the output in a ramdisk so we don't chew up my disk endlessly running this tooling.
+RAMDISK_DIR=/run/shm/ceremonies
+mkdir -p "${RAMDISK_DIR}"
+for ceremonyYear in $(find ./ceremonies/ -maxdepth 1 -type d -printf '%P '); do
+    mkdir -p "${RAMDISK_DIR}/${ceremonyYear}"
+done
+if [ ! -L "ceremony-output" ]; then
+    ln -s "${RAMDISK_DIR}/" ceremony-output
+fi
+
+# Simulate previously-performed ceremonies so we have the keys and certificates
+# available to reference.
+ceremony --config ./ceremonies/2000/root-dst.yaml
+ceremony --config ./ceremonies/2015/root-x1.yaml
+ceremony --config ./ceremonies/2020/root-x2.yaml
+ceremony --config ./ceremonies/2020/root-x2-cross-cert.yaml
+## The zombie cross-sign
+ceremony --config ./ceremonies/2021/root-x1-cross-cert.yaml
 
-# Verify the self-signature on these CSRs.
-openssl req -verify -in int-r3.cross-csr.pem -noout
-openssl req -verify -in int-r4.cross-csr.pem -noout
+
+# Simulating intermediate HSM
+ceremony --config ./ceremonies/2020/r3-key.yaml
+ceremony --config ./ceremonies/2020/r4-key.yaml
+ceremony --config ./ceremonies/2023/r8-key.yaml
+ceremony --config ./ceremonies/2023/r9-key.yaml
+ceremony --config ./ceremonies/2023/r10-key.yaml
+ceremony --config ./ceremonies/2020/e1-key.yaml
+ceremony --config ./ceremonies/2020/e2-key.yaml
+ceremony --config ./ceremonies/2023/e5-key.yaml
```

Comment:
This script should do all of the 2020 work before doing any of the 2023 work, imo. In fact, I think it would be even better for each year's subdirectory to have its _own_ run.sh, and for this one to just call those in order. That way we can have nice separation-of-concerns between different years, and never have to touch lines of code regarding 2023 when we're planning for 2025.